### PR TITLE
G3 29 geneweaver cli client excel parsing tool

### PR DESCRIPTION
**Pull request by bergsalex**
Adds the convert (or cn) commands to help with parsing excel files. This command takes an excel file as input and converts it to the geneweaver batch upload format.

This PR will close G3-29 and G3-30.

```
 gweave alpha parse --help
                                                                
 Usage: gweave alpha parse [OPTIONS] COMMAND [ARGS]...          
                                                                
 Tools and utilities to parse data files for use in Geneweaver. 
 The parse commands help to transform data files in various     
 formats into data files that                                   
 can be uploaded to Geneweaver.                                 
                                                                
╭─ Options ────────────────────────────────────────────────────╮
│ --help          Show this message and exit.                  │
╰──────────────────────────────────────────────────────────────╯
╭─ Commands ───────────────────────────────────────────────────╮
│ cn         Alias for `convert` command.                      │
│ convert    Convert files from one format to another.         │
│ utils                                                        │
╰──────────────────────────────────────────────────────────────╯
```

```
gweave alpha parse convert --help
                                                                
 Usage: gweave alpha parse convert [OPTIONS] FILE_PATH          
                                                                
 Convert files from one format to another.                      
                                                                
╭─ Arguments ──────────────────────────────────────────────────╮
│ *    file_path      PATH  [default: None] [required]         │
╰──────────────────────────────────────────────────────────────╯
╭─ Options ────────────────────────────────────────────────────╮
│ *  --value-header        TEXT      [default: None]           │
│                                    [required]                │
│    --to                  [csv|gw]  [default:                 │
│                                    CovertFileType.BATCH]     │
│    --id-header           TEXT      [default: symbol]         │
│    --help                          Show this message and     │
│                                    exit.                     │
╰──────────────────────────────────────────────────────────────╯
```

This PR is being created now for provisional review since it has gotten quite large. I plan to make the following change prior to merging:

1. Add tests for the remaining utils files that were created / coverage back to _at least_ 70%.
2. Changes to the options for the convert command to move the header choices to the prompting steps.

### CODE REVIEW

**Review by gerring — DISMISSED** [review]
Do there need to be further tests of invalid input files? (xls old versions, xls wrong column layout, xlsx empty file, etc) Checking that the error message is reasonable.

**Review by C-1337ka — APPROVED** [review]

File: src/geneweaver/client/cli/alpha/parse/convert.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    BATCH = "gw"
+
+
+def convert(
+    file_path: Path,
+    value_header: Annotated[str, typer.Option(prompt=True)],
+    to: CovertFileType = CovertFileType.BATCH,
+    id_header: str = "symbol",
+) -> None:
+    """Convert files from one format to another."""
```
Comment (gerring) [inline src/geneweaver/client/cli/alpha/parse/convert.py:31]: Parameter docs needed? as it's not a _ method and public API

File: src/geneweaver/client/cli/alpha/parse/utils.py [hunk 2]
```diff
@@ -61,12 +64,18 @@ def get_metadata(file_path: Path, sheet: Optional[str] = None) -> None:
...
+            raise typer.Exit(code=1) from e
+
+        print_metadata_csv(file_path, metadata)
     elif file_type == "xlsx":
-        sheet_names, sheet_metadata, _, _, n_sheets = _get_metadata_xlsx(
+        sheet_names, sheet_metadata, _, _, n_sheets = xlsx.get_metadata(
             file_path, sheet
         )
-        _print_metadata_xlsx(file_path, sheet_names, sheet_metadata, n_sheets)
+        print_metadata_xlsx(file_path, sheet_names, sheet_metadata, n_sheets)
```
Comment (gerring) [inline src/geneweaver/client/cli/alpha/parse/utils.py:79]: Is it worth having that if type is xls then raise an exception? I assume that is rooted out already.

Comment (gerring) [reply src/geneweaver/client/cli/alpha/parse/utils.py:79]: Or just else: raise an exception.

File: src/geneweaver/client/utils/cli/print/tables.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+from typing import List
+
+from geneweaver.core.types import DictRow
+from rich.console import Console
+from rich.table import Table
+
+console = Console()
+
+
+def print_tabular_data(headers: List[str], rows: List[DictRow]) -> None:
```
Comment (gerring) [inline src/geneweaver/client/utils/cli/print/tables.py:11]: Public method should have parameter type doc.

**Review by desais-jax — APPROVED** [review]

File: src/geneweaver/client/utils/cli/prompt/bool.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+
+def is_bool(field_type: T) -> bool:
+    """Check if the field_type is a bool."""
+    return field_type is bool
+
+
+def prompt_for_bool(field_name: Optional[str] = None, allow_none: bool = False) -> bool:
+    """Prompt the user to enter a boolean value.
+
```
Comment (gerring) [inline src/geneweaver/client/utils/cli/prompt/bool.py:16]: parameter type doc?

**Review by gerring — CHANGES_REQUESTED** [review]
